### ISSUE TITLE
Selectors not returning Elements

### ISSUE THREAD

**Issue by TonyGemenie** [puppeteer/puppeteer#5101]
### Steps to reproduce

**Tell us about your environment:**

* Puppeteer version: v2.0.0
* Platform / OS version: x86_64 / ArchLinux 
* URLs (if applicable): any 
* Node.js version: 12.13

**What steps will reproduce the problem?**

 I've tried a lot of different combinations of selectors but don't seem to be having any luck gaining focus. 

_Please include code that reproduces the issue._

  const browser = await puppeteer.launch();
  const page = await browser.newPage();
  await page.goto('website');
  var data = await page.$eval('a[href|="data:text"]');   
 
**What is the expected result?**
  
   return string representation of the href link

**What happens instead?**

  can't gain focus of any element using a variety of selectors

**Comment by feesler** [comment]
You forgot about 2nd parameter of page.$eval() method. It takes function to run in the page context and return any specific property of selected elements:
`var data = await page.$eval('a[href|="data:text"]', el => el.href);`

**Comment by TonyGemenie** [comment]
I tried with just .$ and used a second term along with other combinations but could never gain focus.  The first day the program ran your syntax without the regex and just a regular selector worked.

**Comment by TonyGemenie** [comment]
In the answer above this I sound like a lunatic.  Problems.  What I meant to say is I used the same syntax as feesler's solution without the regex(the first term) and replaced it with a selector copied from chrome.  It worked well the first day but didn't continue to work.  Then I tried different combinations of single term selectors using .$ without getting focus.  After that I moved on to using different styles of regex's with .$eval without getting focus.  I realize what the problem is but would like a better method for debugging code when I'm not getting focus of an element.
